core/offchain/src/api.rs

```diff
@@ -285,6 +316,72 @@ impl<Storage, KeyProvider> OffchainExt for Api<Storage, KeyProvider> where
 	}
 }
 
+/// Information about the local node's network state.
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub struct NetworkState {
+	peer_id: PeerId,
+	external_addresses: Vec<Multiaddr>,
+}
+
+impl NetworkState {
+	fn new(peer_id: PeerId, external_addresses: Vec<Multiaddr>) -> Self {
+		NetworkState {
+			peer_id,
+			external_addresses,
+		}
+	}
+}
+
+impl From<NetworkState> for OpaqueNetworkState {
+	fn from(state: NetworkState) -> OpaqueNetworkState {
+		let enc = Encode::encode(&state.peer_id.as_bytes().to_vec());
+		let peer_id = OpaquePeerId::new(enc);
+
+		let external_addresses: Vec<OpaqueMultiaddr> = state
+			.external_addresses
+			.iter()
+			.map(|multiaddr| {
+				let e = Encode::encode(&multiaddr.to_string());
+				OpaqueMultiaddr::new(e)
+			})
+			.collect();
+
+		OpaqueNetworkState {
+			peer_id,
+			external_addresses,
+		}
+	}
+}
+
+impl From<OpaqueNetworkState> for NetworkState {
+	fn from(state: OpaqueNetworkState) -> NetworkState {
+		let inner_vec = state.peer_id.0;
+
+		let bytes: Vec<u8> = <Vec<u8>>::decode(&mut &inner_vec[..])
```

Comment:
```suggestion
		let bytes: Vec<u8> = Decode::decode(&mut &inner_vec[..])
```

or 

`let bytes = <<Vec<u8>>::decode(&mut &inner_vec[..])`